### ISSUE TITLE
reproduce ETTm2 univariate results in Table 2

### ISSUE THREAD

**Issue by XBR-1111** [thuml/Autoformer#23]
Thank you very much for sharing the code! 
I can't get univariate prediction results even when I use the scripts. I'm running it with pytorch==1.9.0 and python3.7.
The log is as follow.

Namespace(activation='gelu', batch_size=32, c_out=1, checkpoints='./checkpoints/', d_ff=2048, d_layers=1, d_model=512, data='ETTm2', data_path='ETTm2.csv', dec_in=1, des='Exp', devices='0,1,2,3', distil=True, do_predict=False, dropout=0.05, e_layers=2, embed='timeF', enc_in=1, factor=1, features='S', freq='h', gpu=0, is_training=1, itr=1, label_len=96, learning_rate=0.0001, loss='mse', lradj='type1', model='Autoformer', model_id='ETTm2_96_96', moving_avg=25, n_heads=8, num_workers=10, output_attention=False, patience=3, pred_len=96, root_path='./dataset/ETT-small/', seq_len=96, target='OT', train_epochs=10, use_amp=False, use_gpu=True, use_multi_gpu=False)
Use GPU: cuda:0
>>>>>>>start training : ETTm2_96_96_Autoformer_ETTm2_ftS_sl96_ll96_pl96_dm512_nh8_el2_dl1_df2048_fc1_ebtimeF_dtTrue_Exp_0>>>>>>>>>>>>>>>>>>>>>>>>>>
train 34369
val 11425
test 11425
        iters: 100, epoch: 1 | loss: 0.0966879
        speed: 0.0820s/iter; left time: 872.3655s
        iters: 200, epoch: 1 | loss: 0.1096209
        speed: 0.0755s/iter; left time: 795.7340s
        iters: 300, epoch: 1 | loss: 0.1418100
        speed: 0.0757s/iter; left time: 790.5091s
        iters: 400, epoch: 1 | loss: 0.0920060
        speed: 0.0763s/iter; left time: 789.2272s
        iters: 500, epoch: 1 | loss: 0.1261624
        speed: 0.0763s/iter; left time: 781.5422s
        iters: 600, epoch: 1 | loss: 0.1070141
        speed: 0.0767s/iter; left time: 778.2832s
        iters: 700, epoch: 1 | loss: 0.0975105
        speed: 0.0770s/iter; left time: 773.4665s
        iters: 800, epoch: 1 | loss: 0.0870143
        speed: 0.0773s/iter; left time: 768.7166s
        iters: 900, epoch: 1 | loss: 0.0923501
        speed: 0.0772s/iter; left time: 760.1836s
        iters: 1000, epoch: 1 | loss: 0.1329619
        speed: 0.0775s/iter; left time: 754.6965s
Epoch: 1 cost time: 83.01469206809998
Epoch: 1, Steps: 1074 | Train Loss: 0.1192958 Vali Loss: 0.1495133 Test Loss: 0.1170112
Validation loss decreased (inf --> 0.149513).  Saving model ...
Updating learning rate to 0.0001
        iters: 100, epoch: 2 | loss: 0.1307557
        speed: 0.4428s/iter; left time: 4236.3043s
        iters: 200, epoch: 2 | loss: 0.0746148
        speed: 0.0738s/iter; left time: 698.9521s
        iters: 300, epoch: 2 | loss: 0.0781722
        speed: 0.0739s/iter; left time: 692.0664s
        iters: 400, epoch: 2 | loss: 0.0840582
        speed: 0.0743s/iter; left time: 688.5206s
        iters: 500, epoch: 2 | loss: 0.0945234
        speed: 0.0742s/iter; left time: 679.8758s
        iters: 600, epoch: 2 | loss: 0.1197210
        speed: 0.0742s/iter; left time: 672.6283s
        iters: 700, epoch: 2 | loss: 0.0902257
        speed: 0.0742s/iter; left time: 665.6231s
        iters: 800, epoch: 2 | loss: 0.1256524
        speed: 0.0742s/iter; left time: 658.1316s
        iters: 900, epoch: 2 | loss: 0.1183934
        speed: 0.0743s/iter; left time: 651.0201s
        iters: 1000, epoch: 2 | loss: 0.0817829
        speed: 0.0745s/iter; left time: 645.9785s
Epoch: 2 cost time: 80.55055594444275
Epoch: 2, Steps: 1074 | Train Loss: 0.1038648 Vali Loss: 0.1367796 Test Loss: 0.1242634
Validation loss decreased (0.149513 --> 0.136780).  Saving model ...
Updating learning rate to 5e-05
        iters: 100, epoch: 3 | loss: 0.1129365
        speed: 0.4511s/iter; left time: 3830.8966s
        iters: 200, epoch: 3 | loss: 0.1033758
        speed: 0.0740s/iter; left time: 621.1884s
        iters: 300, epoch: 3 | loss: 0.1275468
        speed: 0.0742s/iter; left time: 615.0512s
        iters: 400, epoch: 3 | loss: 0.0710122
        speed: 0.0745s/iter; left time: 610.2690s
        iters: 500, epoch: 3 | loss: 0.0717602
        speed: 0.0744s/iter; left time: 602.2747s
        iters: 600, epoch: 3 | loss: 0.1175499
        speed: 0.0744s/iter; left time: 595.0589s
        iters: 700, epoch: 3 | loss: 0.1519643
        speed: 0.0749s/iter; left time: 590.9385s
        iters: 800, epoch: 3 | loss: 0.0800418
        speed: 0.0750s/iter; left time: 584.5540s
        iters: 900, epoch: 3 | loss: 0.0746994
        speed: 0.0746s/iter; left time: 573.7547s
        iters: 1000, epoch: 3 | loss: 0.0830479
        speed: 0.0745s/iter; left time: 565.4744s
Epoch: 3 cost time: 80.76482033729553
Epoch: 3, Steps: 1074 | Train Loss: 0.0973873 Vali Loss: 0.1398393 Test Loss: 0.1179208
EarlyStopping counter: 1 out of 3
Updating learning rate to 2.5e-05
        iters: 100, epoch: 4 | loss: 0.0736060
        speed: 0.4432s/iter; left time: 3287.9202s
        iters: 200, epoch: 4 | loss: 0.1116835
        speed: 0.0791s/iter; left time: 579.0581s
        iters: 300, epoch: 4 | loss: 0.0793331
        speed: 0.0791s/iter; left time: 570.8419s
        iters: 400, epoch: 4 | loss: 0.0720609
        speed: 0.0788s/iter; left time: 560.6287s
        iters: 500, epoch: 4 | loss: 0.1089775
        speed: 0.0789s/iter; left time: 553.9732s
        iters: 600, epoch: 4 | loss: 0.0835999
        speed: 0.0786s/iter; left time: 544.0225s
        iters: 700, epoch: 4 | loss: 0.1258804
        speed: 0.0786s/iter; left time: 536.2984s
        iters: 800, epoch: 4 | loss: 0.0553840
        speed: 0.0783s/iter; left time: 526.0710s
        iters: 900, epoch: 4 | loss: 0.0811464
        speed: 0.0788s/iter; left time: 521.5191s
        iters: 1000, epoch: 4 | loss: 0.1213600
        speed: 0.0785s/iter; left time: 511.5148s
Epoch: 4 cost time: 85.34881067276001
Epoch: 4, Steps: 1074 | Train Loss: 0.0949577 Vali Loss: 0.1529772 Test Loss: 0.1448818
EarlyStopping counter: 2 out of 3
Updating learning rate to 1.25e-05
        iters: 100, epoch: 5 | loss: 0.0767112
        speed: 0.4476s/iter; left time: 2840.0901s
        iters: 200, epoch: 5 | loss: 0.0909449
        speed: 0.0783s/iter; left time: 488.9913s
        iters: 300, epoch: 5 | loss: 0.1096216
        speed: 0.0781s/iter; left time: 480.1752s
        iters: 400, epoch: 5 | loss: 0.1170162
        speed: 0.0787s/iter; left time: 475.6964s
        iters: 500, epoch: 5 | loss: 0.0840287
        speed: 0.0778s/iter; left time: 462.4421s
        iters: 600, epoch: 5 | loss: 0.0611459
        speed: 0.0773s/iter; left time: 451.9506s
        iters: 700, epoch: 5 | loss: 0.0679403
        speed: 0.0777s/iter; left time: 446.5334s
        iters: 800, epoch: 5 | loss: 0.0693620
        speed: 0.0783s/iter; left time: 442.0431s
        iters: 900, epoch: 5 | loss: 0.1007061
        speed: 0.0784s/iter; left time: 434.5144s
        iters: 1000, epoch: 5 | loss: 0.0944251
        speed: 0.0783s/iter; left time: 426.3235s
Epoch: 5 cost time: 84.73156476020813
Epoch: 5, Steps: 1074 | Train Loss: 0.0931660 Vali Loss: 0.1550726 Test Loss: 0.1493730
EarlyStopping counter: 3 out of 3
Early stopping
>>>>>>>testing : ETTm2_96_96_Autoformer_ETTm2_ftS_sl96_ll96_pl96_dm512_nh8_el2_dl1_df2048_fc1_ebtimeF_dtTrue_Exp_0<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
test 11425
test shape: (357, 32, 96, 1) (357, 32, 96, 1)
test shape: (11424, 96, 1) (11424, 96, 1)
mse:0.12426342070102692, mae:0.24697716534137726

Similar results are observed with unfixed random seed and itr=3. Are there any other tips for reproducing this work?

**Comment by XBR-1111** [comment]
I can reproduce most of other results in multivariate forecasting. So I was wondering if there's any mistake in the script.

**Comment by zzw-zwzhang** [comment]
Can you provide the parameters of univariate results on Exchange dataset? Thanks!

**Comment by wuhaixu2016** [comment]
Hi, we have already uploaded the univariate scripts for the Exchange dataset.
You can find here: https://github.com/thuml/Autoformer/blob/main/scripts/Exchange_script/Autoformer_univariate.sh

**Comment by XBR-1111** [comment]
Thanks! I will try new scripts later.
